internal/controller/pdb_to_pdbwatcher_controller.go

```diff
@@ -0,0 +1,133 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	types "github.com/paulgmiller/k8s-pdb-autoscaler/api/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// PDBToPDBWatcherReconciler reconciles a PodDisruptionBudget object.
+type PDBToPDBWatcherReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+// Reconcile reads the state of the cluster for a PDB and creates/deletes PDBWatchers accordingly.
+func (r *PDBToPDBWatcherReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	// Fetch the PodDisruptionBudget object based on the reconcile request
+	var pdb policyv1.PodDisruptionBudget
+	err := r.Get(ctx, req.NamespacedName, &pdb)
+	if err != nil {
+		// If the PDB is deleted, we should delete the corresponding PDBWatcher.
+		// First, check if the PDBWatcher exists
+		var pdbWatcher types.PDBWatcher
+		err := r.Get(ctx, req.NamespacedName, &pdbWatcher)
+		if err == nil {
+			// Delete PDBWatcher if PDB is deleted
+			err := r.Delete(ctx, &pdbWatcher)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("unable to delete PDBWatcher: %v", err)
+			}
+			log.Log.Info("Deleted PDBWatcher", "name", req.NamespacedName)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// If the PDB exists, create a corresponding PDBWatcher if it does not exist
+	var pdbWatcher types.PDBWatcher
+	err = r.Get(ctx, req.NamespacedName, &pdbWatcher)
+
+	if err != nil {
```

Comment:
notfound not just error  and return early if found or error because you want your main logic not stuck inside if. (uber style guide has something on this)